smarts/core/remote_agent_buffer.py

```diff
@@ -32,17 +32,23 @@
 
 
 class RemoteAgentBuffer:
-    def __init__(self, zoo_worker_addrs=None, buffer_size=3):
+    def __init__(self, zoo_worker_addrs=None, authkey=b"secret", buffer_size=3):
         """
         Args:
-          buffer_size: Number of RemoteAgents to pre-initialize and keep running in the background, must be non-zero (default: 3)
+            zoo_worker_addrs:
+                List of (ip, port) tuples of Zoo Workers, used to instantiate remote social agents
+            authkey:
+                Authentication key for communication with Zoo Workers
+            buffer_size: 
+                Number of RemoteAgents to pre-initialize and keep running in the background, must be non-zero (default: 3)
         """
         assert buffer_size > 0
 
         self._log = logging.getLogger(self.__class__.__name__)
 
         self._local_zoo_worker = None
         self._local_zoo_worker_addr = None
+        self._authkey = authkey
```

Comment:
It might be useful to add a validation for the `auth_key` here.

Reply:
Type assertion has been added.

```python
assert isinstance(auth_key, bytes), f"Received auth_key of type {type(auth_key)}, but need auth_key of type <class 'bytes'>."
```